06_gpu_and_ml/flan_t5/finetune.py

```diff
@@ -0,0 +1,242 @@
+        xsum_test = load_dataset("xsum", split=f"test[:{size_percentage}%]")
+
+    # Load the whole dataset
+    else:
+        xsum = load_dataset("xsum")
+        xsum_train = xsum["train"]
+        xsum_test = xsum["test"]
+
+    # Load the tokenizer and model
+    tokenizer = AutoTokenizer.from_pretrained(BASE_MODEL)
+    model = AutoModelForSeq2SeqLM.from_pretrained(BASE_MODEL)
+
+    # Replace all padding tokens with a large negative number so that the loss function ignores them in
+    # its calculation
+    padding_token_id = -100
+
+    batch_size = 8
+
+    def preprocess(batch):
+        # prepend summarize: prefix to document to convert the example to a summarization instruction
+        inputs = ["summarize: " + doc for doc in batch["document"]]
+        
+        model_inputs = tokenizer(
+            inputs,
+            max_length=512,
+            truncation=True,
+            padding="max_length"
+        )
+
+        labels = tokenizer(
+            text_target=batch["summary"],
+            max_length=128,
+            truncation=True,
+            padding="max_length"
+        )
+
+        labels["input_ids"] = [[l if l != tokenizer.pad_token_id else padding_token_id for l in label] for label in labels["input_ids"]]
+
+        model_inputs["labels"] = labels["input_ids"]
+        return model_inputs
+
+    tokenized_xsum_train = xsum_train.map(
+        preprocess,
+        batched=True,
+        remove_columns=["document", "summary", "id"]
+    )
+
+    tokenized_xsum_test = xsum_test.map(
+        preprocess,
+        batched=True,
+        remove_columns=["document", "summary", "id"]
+    )
+
+    data_collator = DataCollatorForSeq2Seq(
+        tokenizer,
+        model=model,
+        label_pad_token_id=padding_token_id,
+        pad_to_multiple_of=batch_size,
+    )
+
+    training_args = Seq2SeqTrainingArguments(
+        # Save checkpoints to the mounted volume
+        output_dir=str(VOL_MOUNT_PATH / "model"),
+        per_device_train_batch_size=batch_size,
+        per_device_eval_batch_size=batch_size,
+        predict_with_generate=True,
+        learning_rate=3e-5,
+        num_train_epochs=num_train_epochs,
+        # Save logs to the mounted volume
+        logging_dir=str(VOL_MOUNT_PATH / "logs"),
+        logging_strategy="steps",
+        logging_steps=100,
+        evaluation_strategy="epoch",
+        save_strategy="epoch",
+        save_total_limit=2,
+        load_best_model_at_end=True,
+    )
+
+    trainer = Seq2SeqTrainer(
+        model=model,
+        args=training_args,
+        data_collator=data_collator,
+        train_dataset=tokenized_xsum_train,
+        eval_dataset=tokenized_xsum_test,
+    )
+
+    trainer.train()
+
+    # Save the trained model and tokenizer to the mounted volume
+    model.save_pretrained(str(VOL_MOUNT_PATH / "model"))
+    tokenizer.save_pretrained(str(VOL_MOUNT_PATH / "tokenizer"))
+
+
+# ## Monitoring Finetuning with Tensorboard
+#
+
```

Comment:
Can you add a sentence or two explaining that this is Tensorboard served as a Modal WSGI app?